torch/csrc/utils/tensor_flatten.h

```diff
@@ -37,14 +41,35 @@ struct TensorGroup {
   }
 };
 
-std::vector<TensorGroup> take_tensors(at::TensorList tensors, size_t size_limit);
-void reorder_tensors_like(std::vector<at::Tensor>& tensors, at::TensorList order);
+// Helper function that take a list of tensor by the size limit, if the
+// list of tensors are of different tensor types, each type will be taken by
+// the size limit. In other words, the input tensor list will firstlbe bucketed
+// by tensor types, and then each bucket will be further bucketed by size_limit
```

Comment:
Can you clarify further what _taken by the size limit_ means. I have enough context to know what's going on but can see how a new person reading this doesn't understand what goes on. Can you mention size limit is in bytes, and what an example output is. E.g. `vector{vector{f16}, vector{f16, f16, f16}, vector{f32, f32}}` or something along those lines.

Also some typos.

Reply:
Let me rewrite